### DISCUSSION TITLE
Hardware Flow Control

### DISCUSSION THREAD

**Discussion by teeten** [darkxst/silabs-firmware-builder#68] [Q&A] [answered]
Maybe a bit of a naïve question, but can someone briefly explain what is hardware flow control, and, why is it important (or not) to have a controller with this feature?

Some context:

I am running HA in a docker container on Ubuntu, on a Minisforum N40 fanless mini PC with Intel Celeron N4020 CPU and the SONOFF ZBDongle-P controller (Zigbee only, TI CC2652P chip) plugged in a USB3 port (computer does not have USB2) via a 40cm USB cable (to move the dongle away from the computer). The PC is powered through a mini UPS (Tecnoware FGCERAPLDC252).

I am approaching 50 Zigbee devices (47 right now including bulbs, sensors, relays and buttons from various manufacturers like eWeLink, Ikea or Tuya). I need to add at least 20 more, but it is becoming difficult/impossible to add or re-add devices: ZHA "add device" does not detect anything or stays stuck at the "start interview" step. I already spent a lot of time googling for causes and solutions. I am still not sure if the issue is with the hardware/firmware (devices, controller), or the software (HA/ZHA).

To reduce radio interferences, I have deactivated my router’s 2.4GHz wifi antennas (I use only 5Ghz) and disabled bluetooth on the mini PC.

Now I am considering looking for a better controller device, and wondering if hardware flow control can be a selection criteria. Hence my question above.

**Comment by darkxst** [comment, marked as answer]
Hardware flow control is used by the serial UART connection, to theoretically control flow of data if buffers get full. However in practice it makes little to no difference, the ZBDongle-E without hardware flow control has been very stable and no issues whatsoever due to lacking hardware flow control. On the other hand the ZB-GW04 which has hardware flow control suffers stability issues.

In terms of firmware support, SkyConnect, ZBDongle-E and SMLIGHT SLZB-07 are the best picks and all well tested. There are a few more dongles that will be getting support later on (like Aeotec ZGA008), but not been thoroughly tested yet.
